hive/src/test/java/org/apache/iceberg/hive/HiveCreateReplaceTableTest.java

```diff
@@ -0,0 +1,238 @@
+    exceptionRule.expectMessage("Table already exists: hivedb.tbl");
+
+    Assert.assertFalse("Table should not exist", catalog.tableExists(TABLE_IDENTIFIER));
+
+    // create a table before starting a transaction
+    catalog.createTable(TABLE_IDENTIFIER, SCHEMA, SPEC);
+    Assert.assertTrue("Table should be created", catalog.tableExists(TABLE_IDENTIFIER));
+
+    // should not be possible to start a new create table transaction
+    catalog.newCreateTableTransaction(
+        TABLE_IDENTIFIER, SCHEMA, SPEC, tableLocation, Maps.newHashMap());
+  }
+
+  @Test
+  public void testReplaceTableTxn() {
+    catalog.createTable(TABLE_IDENTIFIER, SCHEMA, SPEC, tableLocation, Maps.newHashMap());
+    Assert.assertTrue("Table should exist", catalog.tableExists(TABLE_IDENTIFIER));
+
+    Transaction txn = catalog.newReplaceTableTransaction(TABLE_IDENTIFIER, SCHEMA, false);
+    txn.commitTransaction();
+
+    Table table = catalog.loadTable(TABLE_IDENTIFIER);
+    Assert.assertEquals("Partition spec should match", PartitionSpec.unpartitioned(), table.spec());
+  }
+
+  @Test
+  public void testReplaceTableTxnTableNotExists() {
+    exceptionRule.expect(NoSuchTableException.class);
+    exceptionRule.expectMessage("No such table: hivedb.tbl");
+
+    Transaction txn = catalog.newReplaceTableTransaction(TABLE_IDENTIFIER, SCHEMA, SPEC, false);
+    txn.updateProperties()
+        .set("prop", "value")
+        .commit();
+    // expect the transaction to fail
+    txn.commitTransaction();
+  }
+
+  @Test
+  public void testReplaceTableTxnTableDeletedConcurrently() {
+    exceptionRule.expect(NoSuchTableException.class);
+    exceptionRule.expectMessage("No such table: hivedb.tbl");
+
+    catalog.createTable(TABLE_IDENTIFIER, SCHEMA, SPEC, tableLocation, Maps.newHashMap());
+    Assert.assertTrue("Table should exist", catalog.tableExists(TABLE_IDENTIFIER));
+
+    Transaction txn = catalog.newReplaceTableTransaction(TABLE_IDENTIFIER, SCHEMA, SPEC, false);
+
+    catalog.dropTable(TABLE_IDENTIFIER);
+
+    txn.updateProperties()
+        .set("prop", "value")
+        .commit();
+
+    // expect the transaction to fail
+    txn.commitTransaction();
+  }
+
+  @Test
+  public void testReplaceTableTxnTableModifiedConcurrently() {
+    Table table = catalog.createTable(TABLE_IDENTIFIER, SCHEMA, SPEC, tableLocation, Maps.newHashMap());
+    Assert.assertTrue("Table should exist", catalog.tableExists(TABLE_IDENTIFIER));
+
+    Transaction txn = catalog.newReplaceTableTransaction(TABLE_IDENTIFIER, SCHEMA, SPEC, false);
+
+    // update the table concurrently
+    table.updateProperties()
+        .set("another-prop", "another-value")
+        .commit();
+
+    txn.updateProperties()
+        .set("prop", "value")
+        .commit();
+    txn.commitTransaction();
+
+    // the replace should still succeed
+    table = catalog.loadTable(TABLE_IDENTIFIER);
+    Assert.assertEquals("Table props should match", "value", table.properties().get("prop"));
```

Comment:
This doesn't test that the other property was replaced, only that the new property is set. I think this assertion should test that the property maps are equal and that only the `"prop" -> "value"` pair is present because `"another-prop"` was overwritten with the previous table state.

Reply:
Added validation above